src/waveform/renderers/waveformrenderbeat.cpp

```diff
@@ -49,53 +63,108 @@ void WaveformRenderBeat::draw(QPainter* painter, QPaintEvent* /*event*/) {
     const double lastDisplayedPosition =
             m_waveformRenderer->getLastDisplayedPosition();
 
-    // qDebug() << "trackSamples" << trackSamples
-    //          << "firstDisplayedPosition" << firstDisplayedPosition
-    //          << "lastDisplayedPosition" << lastDisplayedPosition;
+    // Calculate beat length
+    double beatLength = (60.0 * trackInfo->getSampleRate() / trackInfo->getBpm()) * 2;
```

Comment:
The 2 is samplesPerFrame in a stereo frame, right?
Please use a constant. We have the idea to allow 6 channels one day.

Reply:
Also the 60 should be a named constant.

Reply:
I recognize that I was guessing a little bit while building the formula and was not sure of why the 2 was needed :)

I have replaced the values by constants. Not sure what is the preferred naming convention here.

Anyway, as part of my work on that bug I want to move the logic about beat calculation to the Beats class. Then I mill move all that to there.